tests/test_username_api.py

```diff
@@ -13,9 +13,16 @@
 
 invalid_username = '$very%long{invalid}user(name)'
 
-def generate_random_username(length):
-  return ''.join(random.choice(string.ascii_uppercase + string.ascii_lowercase \
-    + string.digits) for _ in range(length))
+def generate_random_username(length, first_char_letter=False):
+  return ''.join(generate_random_character(length, first_char_letter))
+
+def generate_random_character(length, first_char_letter):
```

Comment:
this should be a separate commit. It is being applied to all of the websites, which is good.

Reply:
what do you mean "It is being applied to all of the websites"?

Reply:
This change, is a separate enhancement, and it is enhancing all sites, not just OpenHub & DeviantArt.
Therefore, it should be (at least) a separate commit in this PR.

You could also do a separate issue & PR for it, as it is not necessary for the rest of your PR.  I can even create a GCI task for it.

Reply:
got it. But can I extract part of changes of a commit?

Reply:
nevermind. I found solution.